### PR TITLE
IOException handling in Stage1Lib.download

### PR DESCRIPTION
**Pull request by tim-zh**
See [failed build](https://circleci.com/gh/cvogt/cbt/489).

Should there be a retry attempt?

### CODE REVIEW

File: stage1/Stage1Lib.scala [hunk 1]
```diff
@@ -43,7 +43,7 @@ class Stage1Lib( val logger: Logger ) extends BaseLib{
 
   def array2hex(padTo: Int, array: Array[Byte]): String = {
     val hex = new java.math.BigInteger(1, array).toString(16)
-    ("0" * (padTo-hex.size)) ++ hex
+    ("0" * (padTo-hex.length)) ++ hex
```
Comment (cvogt) [inline stage1/Stage1Lib.scala]: is there any different between the two?

File: stage1/Stage1Lib.scala [hunk 2]
```diff
@@ -69,10 +69,14 @@ class Stage1Lib( val logger: Logger ) extends BaseLib{
...
-        } finally {
-          stream.close()
+        try {
+          try
+            Files.copy(stream, incomplete, StandardCopyOption.REPLACE_EXISTING)
+          finally
+            stream.close()
+        } catch {
+          case e: IOException =>
+            System.err.println(red("downloading failed ") ++ e.getMessage)
```
Comment (tim-zh) [inline stage1/Stage1Lib.scala]: So what should I do then? Add a retry and print stack trace, or print a stack trace with an info message to mark it as expected error, or leave the code as it is?

File: stage1/Stage1Lib.scala (hunk 1 above)
Comment (tim-zh) [inline stage1/Stage1Lib.scala]: A quote from IntelliJ IDEA's inspection:

> While such calls are legitimate, they require an additional implicit conversion to SeqLike to be made.
Considering the common use cases, calling length on arrays and strings may provide significant advantages.

File: stage1/Stage1Lib.scala (hunk 2 above)
Comment (cvogt) [inline stage1/Stage1Lib.scala]: So my view is: let's not catch the exception and let the jvm show it to the user. But I want this to be a project where things are openly discussed, so I am open to discuss other opinions.

File: stage1/Stage1Lib.scala (hunk 1 above)
Comment (cvogt) [inline stage1/Stage1Lib.scala]: sounds good :)

**Comment by cvogt** [comment]
Btw. re-ran the build successfully. Waiting for a response on my comment.

**Comment by tim-zh** [comment]
PR is closed, since the conclusion is not to catch the exception.

**Comment by cvogt** [comment]
@tim-zh the other minor code improvements would still be nice to merge :)

**Comment by tim-zh** [comment]
@cvogt I'll create another PR as part of #25.

**Pull request closed without merging**